config/config.kinesis.extended.hocon

```diff
@@ -85,11 +85,16 @@
       "backoffPolicy": {
         "minBackoff": 100 milliseconds
         "maxBackoff": 10 seconds
+        "maxRetries": 10
```

Comment:
But do we really want to limit the number of retries ? The plan was that in case of Kinesis glitch, the app keeps the records in-memory until Kinesis is available again and then when it's available again it goes back to normal processing

Reply:
That was a good plan, but now I think we need to change the plan.

We are seeing in production that sometimes the KPL completely halts processing, even when Kinesis is healthy.  It appears that restarting the app fixes the problem.  We don't know exactly why it halts, because the logs from KPL are not great.

I think we still have a lot to learn about KPL in production, and we are just at the start of that journey.  But for now, I think it is better if the app crashes when KPL cannot publish records.  The orchestration (kubernetes or whatever) will immediately restart the application after it crashes, so it self-recovers instead of needing manual intervention.  And we get a exception logged to Sentry.

Reply:
> We are seeing in production that sometimes the KPL completely halts processing, even when Kinesis is healthy.

Too bad! Then your change is great :+1: